native/core/src/execution/planner.rs

```diff
@@ -2610,10 +2610,23 @@ impl From<ExpressionError> for DataFusionError {
 /// modification. This is used to determine if we need to copy the input batch to avoid
 /// data corruption from reusing the input batch.
 fn can_reuse_input_batch(op: &Arc<dyn ExecutionPlan>) -> bool {
-    if op.as_any().is::<ProjectionExec>() || op.as_any().is::<LocalLimitExec>() {
-        can_reuse_input_batch(op.children()[0])
+    if op.as_any().is::<ScanExec>() {
+        // JVM side can return arrow buffers to the pool
```

Comment:
Isn't the JVM returning a buffer to the pool the normal behavior? Native allocates -> FFI -> JVM consumes and releases the buffer?
Also, how is `native_iceberg_compat`  reusing the buffer?